source/SH3/graphics/texture.cpp

```diff
@@ -4,44 +4,69 @@
  *
  *  @copyright 2016  Palm Studios
  *
- *
  *  @date 2-1-2017
  *
  *  @author Jesse Buhagiar
  */
 #include <SH3/graphics/texture.hpp>
 #include <SH3/system/log.hpp>
+#include <SH3/arc/vfile.hpp>
+
+#include <cassert>
+
 
 using namespace sh3_graphics;
 
-std::string texture::load_error::message() const
+//TODO: Scale the texture and then
+void sh3_texture::Load(sh3_arc& mft, const std::string& filename)
 {
-    std::string error;
-    switch(result)
-    {
-    case load_result::SUCCESS:
-        error = "Success";
-        break;
-    case load_result::INVALID_CHUNK:
-        error = "Invalid chunk";
-        break;
-    case load_result::END_OF_FILE:
-        error = "End of file";
-        break;
-    }
-    return error;
-};
+    sh3_texture_header header;
+    sh3_arc_vfile file(mft, filename);
+    sh3_arc_vfile::read_error e;
 
-texture::load_error sh3_graphics::texture::Load(const std::string& filename, sh3_arc& arc)
-{
-    load_error error;
+    std::size_t ret = file.ReadData(&header, sizeof(header), e); // Read in the header
+
+    if(ret != sizeof(header))
+        die("sh3_texture::Load( ): Header only read %d bytes of data! It should have been %d", ret, sizeof(header)); // TODO: Fallback on dummy texture (like Source's purple and black checkboard)
+
+    if(header.bpp == 8)
+        Log(LogLevel::WARN, "WARNING: 8-bit textures are currently unsupported! They WILL look like nothing but pixel garbage.");
+
+    Log(LogLevel::INFO, "Texture info:\n\ttexWidth:%d\n\ttexHeight:%d\n\ttexSize:%d\n\tbpp:%d", header.texWidth, header.texHeight, header.texSize, header.bpp);
+
+    file.Seek(sizeof(header) + header.dataOffset, std::ios_base::beg);
 
-    if(arc.LoadFile(filename, pixbuff) == ARC_FILE_NOT_FOUND)
+    data.resize(header.texSize);
+
+    if(header.texSize == header.texWidth * header.texHeight * header.bpp)
```

Comment:
Do an early return, i.e. instead of
```
if(x)
{
    A;
}
else
{
    B;
    return;
}
```
do
```
if(!x)
{
    B;
    return;
}
A;
````